src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -39,23 +51,55 @@ public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
         $this->joinRelations($queryBuilder, $resourceClass);
     }
 
+    public function getMetadataProperties($resourceClass): array {
+        $properties = [];
+
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $property) {
+            $properties[$property] = $this->propertyMetadataFactory->create($resourceClass, $property);
+        }
+
+        return $properties;
+    }
+
     /**
      * Left joins relations to eager load.
      *
      * @param QueryBuilder $queryBuilder
      * @param string       $resourceClass
      */
-    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass)
+    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass, string $originAlias = 'o', string &$relationAlias = 'a')
     {
-        $classMetaData = $queryBuilder->getEntityManager()->getClassMetadata($resourceClass);
+        $classMetadata = $queryBuilder->getEntityManager()->getClassMetadata($resourceClass);
+        $j = 0;
 
-        foreach ($classMetaData->getAssociationNames() as $i => $association) {
-            $mapping = $classMetaData->associationMappings[$association];
+        foreach ($classMetadata->getAssociationNames() as $i => $association) {
+            $mapping = $classMetadata->associationMappings[$association];
+            $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $association);
 
-            if (ClassMetadataInfo::FETCH_EAGER === $mapping['fetch']) {
-                $queryBuilder->leftJoin('o.'.$association, 'a'.$i);
-                $queryBuilder->addSelect('a'.$i);
+            if (ClassMetadataInfo::FETCH_EAGER !== $mapping['fetch'] || false === $propertyMetadata->isReadableLink()) {
+                continue;
             }
+
+            $method = false === $mapping['joinColumns'][0]['nullable'] ? 'innerJoin' : 'leftJoin';
+
+            $associationAlias = $relationAlias.$i;
+            $queryBuilder->{$method}($originAlias.'.'.$association, $associationAlias);
+            $select = [];
+            $targetClassMetadata = $queryBuilder->getEntityManager()->getClassMetadata($mapping['targetEntity']);
+
+            //@TODO useless at the moment, but Doctrine queries non-necessary relations because addSelect($associationAlias) lazy fetch every association
+            //By using this array of properties, it Lazy fetch association, which are initially in an INNER JOIN (fetches them twice ?!)
+            //If I try to use associationAlias when it has every property, but the properties array when it doesn't it fails because it can not fetch parent association
```

Comment:
This is still an issue, @Simperfit mind taking a look? I think you've a good background with Doctrine maybe you can help figuring this out :/.